docs/wiki/get-started/More-with-CSP.md

```diff
@@ -1,141 +1,176 @@
 
-`struct` is a useful data type for this type of information. In csp, you can use [`csp.Struct`](csp.Struct-API), a higher-level data type that can be defined with type-annotated values.
+A `struct` (in the C programming language) is a useful data type for this type of information. In CSP, you can use [`csp.Struct`](csp.Struct-API), a higher-level data type that can be defined with type-annotated values.
 
 ```python
-class Trade(csp.Struct):
-    price: float
+class Cart(csp.Struct):
+    product: str
     qty: int
-    buy: bool
+    add: bool
+    purchase: bool
 ```
 
 ## Build computing nodes
 
-To calculate volume-weighted averages, you need the cumulative sum of previous trade prices and quantities. Hence, your csp node needs to store stateful information. You can use \[`csp.state`\] to declare stateful variables that bound to the node.
+### Track cart updates
+
+To track the total cart items and price, you need stateful variables that can be updated each time the cart is modified.
+You can use `csp.state` to declare stateful variables that are bound to a node.
+
+> \[!TIP\]
+> By convention, state variables are prefixed with `s_`.
 
-A csp node can return multiple named outputs, denoted as `csp.Outputs` type, created using the `csp.output` function. The individual return values can be accessed with dot notation.
+A CSP node can also return multiple named outputs, denoted as `csp.Outputs` type, created using the `csp.output` function. The individual return values can then be accessed with dot notation.
 
 ```python
 @csp.node
-def vwap(trade: ts[Trade]) -> csp.Outputs(vwap=ts[float], qty=ts[int]):
+def update_cart(event: ts[Cart], discount: ts[float]) -> csp.Outputs(total = ts[float], items = ts[int]):
+    """
+    Track of the cart total and number of items.
+    """
     with csp.state():
-        s_cum_notional = 0.0
-        s_cum_qty = 0
+        s_cart_total = 0.0
+        s_cart_items = 0
+
+    if csp.ticked(event):
+        if event.add:
+            s_cart_total += PRODUCTS[event.product] * event.qty
+            s_cart_items += event.qty
+        else:
+            s_cart_total -= PRODUCTS[event.product] * event.qty
+            s_cart_items += event.qty
 
-    if csp.ticked(trade):
-        s_cum_notional += trade.price * trade.qty
-        s_cum_qty += trade.qty
+    final_total = s_cart_total * discount
 
-        csp.output(vwap=s_cum_notional / s_cum_qty, qty=s_cum_qty)
+    if event.purchase:
+        s_cart_total = 0.0
+        s_cart_items = 0
+
+    csp.output(total=final_total, items=s_cart_items)
 ```
 
+### Track discount applied
+
+To apply a discount if the purchase is made in under a minute, you need to keep track of time from the first update to the cart.
+
+You can do this with an alarm in CSP. An alarm is a new Time Series input bound to a node that ticks at scheduled intervals. It can be created using `csp.alarm()` within an `csp.alarms()` context.
```

Comment:
"ticks at scheduled intervals" is inaccurate, its only scheduled on a call by call basis
can rephrase "An alarm is a new Time Series input bound to a node that can tick at a pre-scheduled time"